File: SQL/Transactions/SQLCommitTransaction.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Namespace SQL

    Public Class SQLCommitTransaction
        Inherits SQL.SQLStatement

        Public Overrides ReadOnly Property SQL() As String
            Get

                Select Case MyBase.ConnectionType
                    Case Database.ConnectionType.SQLServer, Database.ConnectionType.SQLServerCompactEdition
                        Return "COMMIT TRANSACTION"
                    Case Database.ConnectionType.MicrosoftAccess
                        Return "COMMIT TRANSACTION"
                    Case Database.ConnectionType.MySQL
                        Return "COMMIT"
                    Case Database.ConnectionType.Pervasive
                        Return "COMMIT"
                    Case Database.ConnectionType.HyperSQL
                        Return "COMMIT"
                    Case Else
                        Throw New NotImplementedException(Me.ConnectionType.ToString)
                End Select

            End Get
        End Property

    End Class

End Namespace

File: SQL/Transactions/SQLCommitTransaction.cs
// ___________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLCommitTransaction : SQL.SQLStatement
	{
		public override string SQL
		{
			get
			{
				switch (base.ConnectionType)
				{
					case Database.ConnectionType.SQLServer:
					case Database.ConnectionType.SQLServerCompactEdition:
						return "COMMIT TRANSACTION";
					case Database.ConnectionType.MicrosoftAccess:
						return "COMMIT TRANSACTION";
					case Database.ConnectionType.MySQL:
						return "COMMIT";
					case Database.ConnectionType.Pervasive:
						return "COMMIT";
					case Database.ConnectionType.HyperSQL:
						return "COMMIT";
					default:
						throw new NotImplementedException(this.ConnectionType.ToString());
				}
			}
		}
	}
}
